internal/incident/history.go

```diff
@@ -0,0 +1,159 @@
+package incident
+
+import (
+	"fmt"
+	"github.com/icinga/icinga-notifications/internal/event"
+	"github.com/icinga/icinga-notifications/internal/recipient"
+	"github.com/icinga/icinga-notifications/internal/rule"
+	"github.com/icinga/icinga-notifications/internal/utils"
+	"github.com/icinga/icingadb/pkg/types"
+	"time"
+)
+
+// Sync initiates an *incident.IncidentRow from the current incident state and syncs it with the database.
+// Before syncing any incident related database entries, this method should be called at least once.
+// Returns an error on db failure.
+func (i *Incident) Sync() error {
```

Comment:
Not sure why this function is in this file now. Doesn't really have anything to do with history.